site/src/components/roadmap/Roadmap.tsx

```diff
@@ -0,0 +1,220 @@
+  issueUrl: string;
+  text: string;
+}
+
+type RoadmapDataItem = {
+  text: string;
+  startDate: string;
+  targetDate: string;
+  status: string;
+  issueUrl: string;
+};
+
+interface CardViewProps {
+  sortedRoadmapData: RoadmapData[];
+  searchQuery: string;
+}
+
+function sortRoadmapDataByDate(roadmapData: RoadmapData[]): RoadmapData[] {
+  const sortedData = [...roadmapData];
+  sortedData.sort((a, b) => {
+    const startDateA = new Date(a.targetDate);
+    const startDateB = new Date(b.targetDate);
+
+    if (isNaN(startDateA.getTime())) {
+      return 1; // Item A doesn't have a valid date, so it should be placed below item B
+    } else if (isNaN(startDateB.getTime())) {
+      return -1; // Item B doesn't have a valid date, so it should be placed below item A
+    } else {
+      return startDateA.getTime() - startDateB.getTime();
+    }
+  });
+  return sortedData;
+}
+
+function RoadmapCard(props: CardProps) {
+  return (
+    <Card
+      className={styles.spacing}
+      style={{ width: "250px", marginBottom: "15px", minHeight: "unset" }}
+      {...props}
+    />
+  );
+}
+
+export const Roadmap = ({ title, children }: RoadmapProps) => {
+  const [jsonData, setJsonData] = useState("");
+  const [roadmapData, setRoadmapData] = useState<any[]>([]);
+  const sortedRoadmapData = sortRoadmapDataByDate(roadmapData);
+  const [searchQuery, setSearchQuery] = useState("");
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        //gets the data
+        const response = await fetch("/api/roadmap");
+        const responseData = await response.json();
+        const jsonString = JSON.stringify(responseData, null, 2);
+        setJsonData(jsonString);
+
+        const items =
+          responseData?.data?.organization?.repository?.projectV2?.items?.nodes;
+
+        //creates an array of objects with data from github
+        const extractedData: RoadmapDataItem[] = items?.map((item: any) => {
+          const fieldValueNodes = item?.fieldValues?.nodes;
+          const text = getFieldValueByName(fieldValueNodes, "Title");
+          const startDate = getFieldValueByName(fieldValueNodes, "Start Date");
+          const targetDate = getFieldValueByName(
+            fieldValueNodes,
+            "Target Date"
+          );
+          const status = getFieldValueByName(fieldValueNodes, "Status");
+          const issueUrl = item?.content?.url;
+
+          return { text, startDate, targetDate, status, issueUrl };
+        });
+
+        //sets the data to retrieved data or null
+        setRoadmapData(extractedData || []);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const getFieldValueByName = (fieldValueNodes: any[], fieldName: string) => {
+    const fieldValueNode = fieldValueNodes?.find(
+      (node: any) => node?.field?.name === fieldName
+    );
+    return (
+      fieldValueNode?.text || fieldValueNode?.date || fieldValueNode?.name || ""
```

Comment:
this function is somewhat magical, you say I want field X and then it returns text, date or name, dependent on what is set on the node

It could be fragile, I might go for something more explicit